### PR TITLE
feat(commands): add github command

### PR DESCRIPTION
**Pull request by vit0rr**
Github command 

CleanShot 2024-06-11 at 17 29 56

### CODE REVIEW

File: src/wweb/middleware/middleware.ts [hunk 1]
```diff
@@ -20,6 +20,8 @@ export const middleware = (
   return async (msg: Message) => {
     // if onlyOwner is true, the message will only be processed if it is sent by the owner,
     // otherwise it will only be processed if it is not sent by the owner
+    if (process.env.NODE_ENV === 'development') return next(msg);
```
Comment (NicolasLopes7) [inline src/wweb/middleware/middleware.ts:23]: shall we short-circuit credit checks too? https://github.com/vinibgoulart/wpp-playground/pull/31/files#diff-70404c349283f04f411c8c76bf6d1eb75523979d0be19c18fc539179a90ec39aR35

Comment (vinibgoulart) [inline src/wweb/middleware/middleware.ts:23]: move this to here: https://github.com/vinibgoulart/wpp-playground/pull/31/files#diff-70404c349283f04f411c8c76bf6d1eb75523979d0be19c18fc539179a90ec39aR29
image

only for message owner middleware

File: src/wweb/onMessage/commands.ts [hunk 2]
```diff
@@ -13,8 +13,19 @@ import quoteMessage from './meme/quoteMessage';
...
 
-const COMMANDS = {
+type Commands = {
+  [key: string]: {
+    name: string;
+    description?: string;
+    example?: string;
+    action: (msg: Message) => Promise<unknown>;
+  };
+}
```
Comment (NicolasLopes7) [inline src/wweb/onMessage/commands.ts:26]:
```suggestion
type Commands = Record<string, {
    name: string;
    description?: string;
    example?: string;
    action: (msg: Message) => Promise<unknown>;
  }>
```

Comment (NicolasLopes7) [reply src/wweb/onMessage/commands.ts:26]: we can might define the commands as well to be fully type-safe

File: src/wweb/middleware/middleware.ts (hunk 1 above)
Comment (vinibgoulart) [reply src/wweb/middleware/middleware.ts:23]:
```tsx
} else {
	const isDevelopment = process.env.NODE_ENV === 'development';
      if (msg.fromMe && !isDevelopment) {
        return;
      }
    }
```

File: src/wweb/onMessage/commands.ts (hunk 2 above)
Comment (vinibgoulart) [reply src/wweb/onMessage/commands.ts:26]: yes, we can do it in another pull request, it is a good improvement

File: src/wweb/middleware/middleware.ts (hunk 1 above)
Comment (daniloab) [reply src/wweb/middleware/middleware.ts]: avoid else

Comment (vit0rr) [reply src/wweb/middleware/middleware.ts]: Why?

**Pull request merged**